fhir-model/src/main/java/com/ibm/fhir/model/path/FHIRPathType.java

```diff
@@ -347,6 +348,20 @@
         metamodelTypeMap.put(SimpleTypeInfo.class, SYSTEM_SIMPLE_TYPE_INFO);
         return metamodelTypeMap;
     }
+    
+    private static Map<Class<?>, FHIRPathType> buildJavaTypeMap() {
+        Map<Class<?>, FHIRPathType> javaTypeMap = new HashMap<>();
+        javaTypeMap.put(java.lang.Boolean.class, SYSTEM_BOOLEAN);
+        javaTypeMap.put(java.lang.Integer.class, SYSTEM_INTEGER);
+        javaTypeMap.put(java.math.BigDecimal.class, SYSTEM_DECIMAL);
+        javaTypeMap.put(java.lang.String.class, SYSTEM_STRING);
+        javaTypeMap.put(java.time.LocalTime.class, SYSTEM_TIME);
+        javaTypeMap.put(java.time.ZonedDateTime.class, SYSTEM_DATE_TIME);
+        javaTypeMap.put(java.time.LocalDate.class, SYSTEM_DATE_TIME);
+        javaTypeMap.put(java.time.YearMonth.class, SYSTEM_DATE_TIME);
+        javaTypeMap.put(java.time.Year.class, SYSTEM_DATE_TIME);
+        return javaTypeMap;
```

Comment:
Should this be an unmodifiable map? 

how does it differ from buildTypeMap?  does buildTypeMap need to respond with this map?

Reply:
Changed all maps and sets to their "unmodifiable" counterparts. There are 3 distinct Map<Class<?>, FHIRPathType> maps. One holds FHIR model class to FHIRPathType mapping, one holds FHIRPath metamodel class to FHIRPathType mapping and one holds java types (used in the model) to their FHIRPath type equivalent.